### ISSUE TITLE
Hooking for Talk Group sync?

### ISSUE THREAD

**Issue by Dubidubiduu** [nextcloud/user_sql#178]
Hey there,

can you have a look into https://github.com/nextcloud/spreed/issues/7039?

Is there anything I can check? From admin backend the user is in the cloud and has the correct groups...

Best

**Comment by ovesco** [comment]
Hey,

I have the exact same problem. It is complicated as it depends on what database system you have, if there is a possibility to setup a listener and so on.

In my case I manually forked this library and implemented a small controller exposing a route which triggers the group listeners, called by my "main" system (where i manage users and groups) when a change is detected. I can help you if needed.

**Comment by Dubidubiduu** [comment]
Thanks for this answer.

Maybe you can link to this fork, for me and others.

But actually I don't want to do something for my own or adopt your way, because I think it just worked in the past?!

I get the data from a wordpress/civicrm VIEW at another server. Nextcloud itself runs on mariadb, sync works, it's just this Talk Group syny that doesn't.

**Comment by ovesco** [comment]
Yes that's because when Talk creates a group discussion, it actually retrieves the various members of the group at the creation moment and adds them to the discussion.

It also listens to internal nextcloud events (including user added/removed from a group) so that it can update discussion members. Those events are dispatched by Nextcloud when we add/remove a member from a group. The problem is that we'd need the same features from Mariadb, IE triggering a PHP hook when it detects a change in your tables which is complicated...

My fork is the following: https://github.com/sauvabelin/user_sql
You'll be more interested by the following controller (which exposes, in my case, a single route to trigger those events): https://github.com/sauvabelin/user_sql/blob/master/lib/Controller/GroupChangeController.php
